packages/web3-providers-http/src/index.ts

```diff
@@ -1,142 +1,182 @@
+    private _connected = false;
 
-    constructor(options: ProviderOptions) {
-        super(options);
-        this._httpClient = Web3ProvidersHttp._createHttpClient(
-            options.providerUrl
-        );
+    web3Client: string;
+
+    constructor(web3Client: string) {
+        super();
+        this._httpClient = Web3ProvidersHttp._createHttpClient(web3Client);
+        this.web3Client = web3Client;
+        this._connectToClient();
     }
 
-    private static _validateProviderUrl(providerUrl: string): boolean {
+    /**
+     * Determines whether {web3Client} is a valid HTTP client URL
+     *
+     * @param web3Client To be validated
+     * @returns true if valid
+     */
+    private static _validateClientUrl(web3Client: Web3Client): boolean {
         try {
-            return (
-                typeof providerUrl !== 'string' ||
-                /^http(s)?:\/\//i.test(providerUrl)
-            );
+            return typeof web3Client === 'string'
+                ? /^http(s)?:\/\//i.test(web3Client)
+                : false;
         } catch (error) {
-            throw Error(`Failed to validate provider string: ${error.message}`);
+            throw Error(`Failed to validate client url: ${error.message}`);
         }
     }
 
-    private static _createHttpClient(baseUrl: string): AxiosInstance {
+    /**
+     * Creates axios instance from {web3Client} URL
+     *
+     * @param web3Client Client URL to send requests to
+     * @returns AxiosInstance
+     */
+    private static _createHttpClient(web3Client: Web3Client): AxiosInstance {
         try {
-            if (!Web3ProvidersHttp._validateProviderUrl(baseUrl))
+            if (!Web3ProvidersHttp._validateClientUrl(web3Client))
                 throw Error('Invalid HTTP(S) URL provided');
-            return axios.create({ baseURL: baseUrl });
+            return axios.create({ baseURL: web3Client as string });
         } catch (error) {
             throw Error(`Failed to create HTTP client: ${error.message}`);
         }
     }
 
-    setProvider(providerUrl: string) {
+    /**
+     * Checks if connection to client is possible by requesting
+     * client's chainId
+     */
+    private async _connectToClient() {
         try {
-            this._httpClient = Web3ProvidersHttp._createHttpClient(providerUrl);
-            super.providerUrl = providerUrl;
+            const chainId = await this._getChainId();
+            this.emit(Web3ProviderEvents.Connect, { chainId });
+            this._connected = true;
+
+            // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md#chainchanged-1
+            if (
+                this._clientChainId !== undefined &&
+                chainId !== this._clientChainId
+            ) {
+                this.emit(Web3ProviderEvents.ChainChanged, chainId);
+            }
+            this._clientChainId = chainId;
         } catch (error) {
-            throw Error(`Failed to set provider: ${error.message}`);
```

Comment:
When client fails to set a provider,  `this._connected = false`

Reply:
I think you're right, and the disconnected event should be emitted, but I want to include this in the broader error handling issue: #4203

Reply:
I don't think it's necessary to do that here since `_connectToClient` is calling `_getChainId` which is [using the _request method](https://github.com/ChainSafe/web3.js/blob/3428041a7c34326410d9ccb2bc123119c8f03111/packages/web3-providers-http/src/index.ts#L93) that will already set [_connected = false](https://github.com/ChainSafe/web3.js/blob/3428041a7c34326410d9ccb2bc123119c8f03111/packages/web3-providers-http/src/index.ts#L173) and emit the `Web3ProviderEvents.Disconnect` event